### PR TITLE
new fields in diff-array cause undefined access error

### PR DESCRIPTION
**Pull request by mxab**
I wrote a test that show cases the problem and a fix that seems to work

### CODE REVIEW

File: lib/diff-array.js [hunk 1]
```diff
@@ -191,6 +200,9 @@ var handleDeepProperties = function(object, paths, cb) {
     var currentObject = object;
 
     for (var i = 0; i < props.length - 1; i++) {
+      if(_.isUndefined(currentObject[props[i]])){
```
Comment (Urigo) [inline lib/diff-array.js]: Looks like this can go away and your other change solves this

Comment (mxab) [inline lib/diff-array.js]: how do you mean? My first try was only the EJSON part in line 138 but than I had still an error that the currentObject was undefined when it tried to apply the diff changes
you can see it when you run the spec and incrementally add the changes

Comment (Urigo) [inline lib/diff-array.js]: Strange, the test passed when I ran it without that line

Comment (mxab) [inline lib/diff-array.js]: Hmm okey that's strange.. I'll re-run it ...

Comment (mxab) [inline lib/diff-array.js]: Hmm if I remove the three lines in the handleDeepProperties for loop i get
```
TypeError: Cannot read property 'images' of undefined
    at http://localhost:3000/packages/urigo_angular.js?8ab7a0c439daa219c61958486f0f9d098d902f6a:573:36
    at Object.forEach (http://localhost:3000/packages/angular_angular.js?c4420bc04009b877e624e5f64e10fef52b19a1fe:380:20)
    at handleDeepProperties (http://localhost:3000/packages/urigo_angular.js?8ab7a0c439daa219c61958486f0f9d098d902f6a:565:11)
    at deepCopyObjectChanges (http://localhost:3000/packages/urigo_angular.js?8ab7a0c439daa219c61958486f0f9d098d902f6a:585:3)
    at Object.<anonymous> (http://localhost:3000/packages/local-test_urigo_angular.js?d7fce7d24c54fcf5502ee92c221fec2bb37a8d14:497:9)
    at attemptSync (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:2286:24)
    at QueueRunner.run (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:2274:9)
    at QueueRunner.execute (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:2260:10)
    at Spec.Env.queueRunnerFactory (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:1096:35)
    at Spec.execute (http://localhost:3000/packages/sanjo_jasmine.js?ca8c9ae4782601ea2ea1c5a1b945c26fc83633ba:845:10)
```

during the _should copy new values_ test

Comment (Urigo) [inline lib/diff-array.js]: OK, I've just pushed a few fixes to a new branch (was suppose to be unrelated to your changes) here: https://github.com/Urigo/angular-meteor/commit/9ad36f8784abd7669add6918aa5deec52dfbe37b

Can you have a short look to make sure we are ok and it's not related?
thanks

Comment (Urigo) [inline lib/diff-array.js]: https://github.com/Urigo/angular-meteor/pull/446

**Comment by Urigo** [comment]
OK, I've merged the other change, so can you rebase again?
thanks!

**Comment by mxab** [comment]
sorry i overlooked the part with the rebase, I just merged in your changes :(

**Comment by Urigo** [comment]
@mxab awesome!

**Pull request merged**